### PR TITLE
fix: correctly advertise minimum supported version

### PR DESCRIPTION
**Pull request by aignas**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [X] Bugfix
- [ ] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [X] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

We don't have CI tests that would help us avoid regressing the minimum supported version. This also means that `py_proto_dependency` addition broke the support for bazel <5 and we should think of better ways to support bazel 4 LTS if this is a requirement.

Issue Number: #1056

## What is the new behavior?

Test all examples with the minimum supported version, which is now set to `bazel 5.4.0`. Feel free to set this to something lower.

## Does this PR introduce a breaking change?

This may potentially break users, who expect the latest `rules_python` to work with a bazel version older than `5.4.0`. I'll do a few more tests to check what the actual lowest version is.

### CODE REVIEW

File: python/private/common.bzl [hunk 1]
```diff
@@ -0,0 +1,5 @@
+"""Common constants for the internal implementation of rules_python.
```
[the remark this answers is not in the document]
Comment (aignas) [reply python/private/common.bzl]: Thanks, done.

File: python/versions.bzl [hunk 2]
```diff
@@ -15,6 +15,9 @@
 """The Python versions we use for the toolchains.
 """
 
+# NOTE: Keep in sync with .bazelci/presubmit.yml
+MINIMUM_BAZEL_VERSION = "5.4.0"
```
Comment (rickeylev) [inline python/versions.bzl:19]: Err, sorry, I gave you the wrong path: top-level "version.bzl" file

Comment (aignas) [reply python/versions.bzl]: Ah, yes, sorry for missing it. Done.

**Comment by rickeylev** [comment]
CI is failing -- I think I encountered the same problem with a recent PR, very cryptic error. I think the target under docs/BUILD.bazel needs to be updated? It needs to have the version.bzl file added to it.

**Comment by aignas** [comment]
> CI is failing -- I think I encountered the same problem with a recent PR, very cryptic error. I think the target under docs/BUILD.bazel needs to be updated? It needs to have the version.bzl file added to it.

It took me a second to understand that it is the strange changes I saw in #1064 😆 .

**Review by f0rmiga — APPROVED** [review]

**Review by rickeylev — APPROVED** [review]

**Pull request merged**